Sources/SwiftLexicalLookup/LookupName.swift

```diff
@@ -0,0 +1,123 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+@_spi(Experimental) public enum LookupName {
+  /// Identifier associated with the name.
+  /// Could be an identifier of a variable, function or closure parameter and more
+  case identifier(IdentifiableSyntax, accessibleAfter: AbsolutePosition?)
+  /// Declaration associated with the name.
+  /// Could be class, struct, actor, protocol, function and more
+  case declaration(NamedDeclSyntax, accessibleAfter: AbsolutePosition?)
```

Comment:
I expect that we'll need another case here eventually for "implicit" names, such as the `error` introduced in

```swift
do {
  try something()
} catch { // the catch clause introduces the name `error`.
}
```

This can, of course, come later.

Reply:
Thanks for the suggestion. I'll introduce implicit names with the next PR.